scripts/compile.sh

```diff
@@ -73,28 +90,37 @@ build_llvm() {
     fi
 
     git checkout ${LLVM_HASH}
+    [ -z "$CLEANUP" ] || rm -rf build
 
-    dylib=OFF
-    if [ "$DYN_LINK" = 'true' ]; then 
-        dylib=ON
-    fi
+    [ "$DYN_LINK" = "OFF" ] && CXX_FLAGS="-fvisibility=hidden"
 
-    [ -z "$CLEANUP" ] || rm -rf build
     cmake -G Ninja llvm -B build \
-        -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS_DEBUG="-g -O0" \
-        -DLLVM_ENABLE_ASSERTIONS=true -DLLVM_ENABLE_PROJECTS="mlir"\
-        -DLLVM_TARGETS_TO_BUILD="X86" -DLLVM_INSTALL_UTILS=true \
-        -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DLLVM_INSTALL_GTEST=ON \
-        -DLLVM_BUILD_LLVM_DYLIB=$dylib -DLLVM_LINK_LLVM_DYLIB=$dylib
-    cmake --build build
+        -DCMAKE_BUILD_TYPE=$BUILD_TYPE \
+        -DCMAKE_CXX_FLAGS="$CXX_FLAGS" \
+        -DCMAKE_CXX_FLAGS_DEBUG="-g -O0" \
+        -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+        -DLLVM_ENABLE_ASSERTIONS=ON \
+        -DLLVM_ENABLE_PROJECTS="mlir"\
+        -DLLVM_TARGETS_TO_BUILD="X86" \
+        -DLLVM_BUILD_LLVM_DYLIB=$DYN_LINK \
+        -DLLVM_LINK_LLVM_DYLIB=$DYN_LINK \
+        -DLLVM_INCLUDE_RUNTIMES=OFF \
+        -DLLVM_INCLUDE_EXAMPLES=OFF \
+        -DLLVM_INCLUDE_TESTS=OFF \
```

Comment:
Would this prevent us from running llvm's mlir checks?

Reply:
The utils like lit, FileCheck, etc. are enabled